src/clipper.rs

```diff
@@ -86,3 +86,61 @@ pub fn clip_rect_pos_uv(pos: &Rect<f32>, uv: &Rect<f32>, clip_rect: &Rect<f32>)
         }
     })
 }
+
+pub fn clip_out_rect_pos_uv(pos: &Rect<f32>, uv: &Rect<f32>, clip_rect: &Rect<f32>)
+                            -> Vec<ClipRectResult> {
+    let clip_rect = match pos.intersection(clip_rect) {
+        Some(clip_rect) => clip_rect,
+        None => return vec![ClipRectResult::from_rects(pos, uv)],
+    };
+
+    // FIXME(pcwalton): Clip the u and v too.
+    let mut result = vec![];
```

Comment:
(My clipping PR has some reusable buffers that could be used here to be even more efficient)